docs/userguide_3.0.md

```diff
@@ -1,7 +1,169 @@
-# TiSpark Configurations
+# TiSpark (version >= 3.0) User Guide
 
-> The configurations in the table below can be put together with `spark-defaults.conf` or passed in the same way as other Spark configuration properties.
+> **Note:**
+>
+> This is a user guide for TiSpark version >= 3.0 which is compatible with spark 3.x. If you are using Spark version earlier than 3.0, refer to [Document for TiSpark 2.x](./userguide_2.0.md).
+> 
+> TiSpark >= 3.0 is developed with Spark catalog plugin. If you want to use TiSpark without catalog plugin mode, please turn to TiSpark 2.5.
+
+- [Requirements](#requirements)
+- [Recommended deployment configurations](#recommended-deployment-configurations)
+- [Getting Started](#getting-started)
+  + [Start spark-shell](#start-spark-shell)
+  + [Get TiSpark version](#get-tispark-version)
+  + [Read with TiSpark](#read-with-tispark)
+  + [Write with TiSpark](#write-with-tispark)
+  + [Write With JDBC DataSource](#write-with-jdbc-datasource)
+  + [Delete with TiSpark](#delete-with-tispark)
+  + [Use with other data source](#use-with-other-data-source)
+- [Configuration](#configuration)
+  + [TLS Configuration](#tls-configuration)
+  + [Log4j Configuration](#log4j-configuration)
+  + [Time Zone Configuration](#time-zone-configuration)
+- [Features](#features)
+  + [Expression Index](#expression-index)
+  + [TiFlash](#tiflash)
+  + [Partition Table support](#partition-table-support)
+  + [Other Features](#other-features)
+- [Statistics information](#statistics-information)
+- [FAQ](#faq)
+
+# Requirements
+
++ TiSpark supports Spark >= 2.3.x, but does not support any Spark versions earlier than 2.3.
++ TiSpark requires JDK 1.8+ and Scala 2.11/2.12.
```

Comment:
I think only 1.8, we haven't tested 1.11, 1.17.

Reply:
yes